src/nadzoring/network_base/whois_lookup.py

```
@@ -44,43 +44,36 @@ def _is_ip(target: str) -> bool:
+        TimeoutExpired: The WHOIS query exceeded the timeout period.
+        CalledProcessError: The whois command returned a non-zero exit code.
     """
+    if timeout_config is None:
+        timeout_config = TimeoutConfig()
+
     os_name: str = system()
     encoding: Literal["cp866", "utf-8"] = "cp866" if os_name == "Windows" else "utf-8"
 
-    try:
+    with timeout_context(timeout_config):
         return check_output(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)